envoy/network/transport_socket.h

```diff
@@ -16,9 +16,11 @@
 
 #include "absl/types/optional.h"
 
+#ifdef ENVOY_ENABLE_QUIC
 namespace quic {
```

Comment:
This probably ought to also be replaced with `#include "quiche/quic/core/crypto/quic_crypto_client_config.h"` instead of a forward declaration. I don't feel strongly either way.

Reply:
I'd be inclined to leave it as is but actually doing the include would be fine too.